src/core/qgscoordinatereferencesystem.cpp

```diff
@@ -2584,3 +2584,70 @@ void QgsCoordinateReferenceSystem::invalidateCache( bool disableCache )
   }
   sCrsStringLock()->unlock();
 }
+
+// invalid < regular < user
+bool operator> ( const QgsCoordinateReferenceSystem &c1, const QgsCoordinateReferenceSystem &c2 )
+{
+  if ( c1.d == c2.d )
+    return false;
+
+  if ( !c1.d->mIsValid && !c2.d->mIsValid )
+    return false;
+
+  if ( !c1.d->mIsValid && c2.d->mIsValid )
+    return false;
+
+  if ( c1.d->mIsValid && !c2.d->mIsValid )
+    return true;
+
+  const bool c1IsUser = c1.d->mSrsId >= USER_CRS_START_ID;
+  const bool c2IsUser = c2.d->mSrsId >= USER_CRS_START_ID;
+
+  if ( c1IsUser && !c2IsUser )
+    return true;
+
+  if ( !c1IsUser && c2IsUser )
+    return false;
+
+  if ( !c1IsUser && !c2IsUser )
+    return c1.d->mAuthId > c2.d->mAuthId;
+
+  return c1.toWkt( QgsCoordinateReferenceSystem::WKT_PREFERRED ) > c2.toWkt( QgsCoordinateReferenceSystem::WKT_PREFERRED );
+}
+
+bool operator< ( const QgsCoordinateReferenceSystem &c1, const QgsCoordinateReferenceSystem &c2 )
+{
+  if ( c1.d == c2.d )
+    return false;
+
+  if ( !c1.d->mIsValid && !c2.d->mIsValid )
+    return false;
+
+  if ( c1.d->mIsValid && !c2.d->mIsValid )
+    return false;
```

Comment:
should this be true, if we're using "invalid < regular < user"?

Reply:
/me is writing some tests, that's better than a discussion in theory